websites/N/namu.wiki/presence.ts

```diff
@@ -0,0 +1,189 @@
+validateContributeUrl = /\/contribution\/(.+)\/(.+)\/(.+)/;
+
+let currentPage = document.location.pathname, currentTime = Date.now();
+presence.on("UpdateData", async () => {
+  const privacy = await presence.getSetting("privacy"),
+  showTimestamp = await presence.getSetting("show_timestamp"),
+  presenceData: PresenceData = {
+    largeImageKey: "namu"
+  };
+
+  if (document.location.hostname !== "namu.wiki") return;
+  const path = document.location.pathname,
+        params = document.location.search,
+        parsedUrl = path.split("/"); // It's a very bad design, but they have a slash document.
+  if (parsedUrl === null) return;
+
+  /**
+   * 
+   * Setting Details & State
+   * 
+   */
+  let privateMode = false;
+  const action = parsedUrl[1], details = boardTypeMapping[action];
+  presenceData.details = ((details === undefined) ? "Unknown Action" : details);
+
+  let page;
+  /* View Contribute */
+  if ((page = validateContributeUrl.exec(path))) {
+    if (page[1] === "author") {
+      page = `User: ${page[2]}`;
+    } else {
+      privateMode = true;
+      page = "IP User";
+    }
+
+  /* View Membership */
+  } else if ((page = validateMembershipUrl.exec(path))) {
+    privateMode = true; // Privacy
+    presenceData.details = "Member Page";
+    page = membersMapping[page[1]];
+
+  /* Searching */
+  // /Search?q=(search)
+  } else if (action === "Search") {
+    page = getParam(params, 'q');
+    if (page.length === 0) page = "Blank Query...";
+
+  /* View Recent Discuss History */
+  } else if (action === "RecentDiscuss")
+    page = discussMapping[getParam(params, 'logtype')];
+
+  /* Recent Changes History */
+  else if (action === "RecentChanges")
+    page = changesMapping[getParam(params, 'logtype')];
+
+  /* View Discuss Thread */
+  else if (action === "thread" || action === "edit_request")
+    page = document.querySelector(`h1 > a`).textContent; // H1 Tag Only one
+
+  /* Other */
+  else if (details !== undefined) page = decodeURI(path.substring(`/${action}/`.length));
+  else page = null;
+
+  if (page === null || page === undefined) page = "";
+  presenceData.state = ((page.length > 128) ? `${page.slice(0, 120)}...` : page);
+
+  /**
+   * 
+   * Login Status
+   * 
+   */
+  if (details !== undefined) {
+    const members = document.querySelectorAll(`#app > div > div > nav > ul[class=r] > li > div > div > div`);
+    if (members[1].textContent.indexOf("Please login!") === -1) {
+      presenceData.smallImageKey = "user";
+      if (!privacy) presenceData.smallImageText = members[0].textContent;
+    }
+  }
+
+  /**
+   * 
+   * Set Time
+   * 
+   */
+  if (currentPage !== path) {
+    currentPage = path;
+    currentTime = Date.now();
+  }
+
+  /**
+   * 
+   * Apply
+   * 
+   */
+  if (presenceData.details === null) {
+    presence.setTrayTitle();
+    presence.setActivity();
+  } else {
+    presenceData.buttons = (privateMode ? null : [{ label: "View Page", url: document.location.href }]);
```

Comment:
Delete it - because it can be shown on homepage and we don't allow it. If you really need buttons add them to subpages.

Reply:
Could you tell me what the subpage is because I am not good at English 😢

Reply:
Subpage - https://en.wikipedia.org/wiki/Subpage